src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -454,13 +454,22 @@ bool Creature::UpdateEntry(uint32 Entry, const CreatureData* data, bool changele
     SetAttackTime(RANGED_ATTACK, cInfo->RangeAttackTime);
 
     uint32 previousHealth = GetHealth();
+    uint32 previousMaxHealth = GetMaxHealth();
     uint32 previousPlayerDamageReq = m_PlayerDamageReq;
 
     SelectLevel(changelevel);
     if (previousHealth > 0)
     {
         SetHealth(previousHealth);
-        m_PlayerDamageReq = previousPlayerDamageReq;
+
+        if (previousMaxHealth > GetMaxHealth())
```

Comment:
`if (previousMaxHealth && previousMaxHealth > GetMaxHealth())`

Better to avoid division by zero in some edge-cases

Reply:
Can it even be null? Would mean that HP doesnt even exist?

Reply:
> Can it even be null? Would mean that HP doesnt even exist?

nope, we could even use `> 0`